### DISCUSSION TITLE
GUI for Genesis

### DISCUSSION THREAD

**Discussion by mayura-andrew** [Genesis-Embodied-AI/genesis-world#193] [Ideas]
Are you guys working on GUI for this ? If yes I'm happy to contribute :)

**Comment by Prier** [comment]
I am also very interested in a GUI, or at least the possibility of interaction with the scene.

**Comment by zhouxian** [comment]
We do need a GUI!!! We didn't have any manpower to do this, and any contribution is welcom!

**Comment by lange4531** [comment]
I'd also be interested in writing a GUI along with anyone else who's interested!

**Reply by mayura-andrew** [reply]
Cool ! We can start :)

**Reply by lange4531** [reply]
It would be fun to have an ad-hoc team for this and get started rather soon :D If anyone else is interested let us know!

**Reply by Excidos** [reply]
I've got a basic Streamlit UI with Ollama integrated to help create the parameters for my scene based on a query.

**Reply by Prier** [reply]
I can help out as well.

**Reply by aakash-ram-chandran** [reply]
I have used isaac sim GUI and know few new ideas and good things. I can also help out. Please let me know how to get start on it or how can i help in this.

**Reply by mayura-andrew** [reply]
how we are going to start this? could we access this genesis through API ?

**Reply by shuquan** [reply]
I can help too. :)

**Reply by JocelynVelarde** [reply]
> I've got a basic Streamlit UI with Ollama integrated to help create the parameters for my scene based on a query. 

Hey @Excidos would you be interested in working on the streamlit together?

**Comment by 00make** [comment]
# ![Genesis GUI](https://github.com/user-attachments/assets/6ea0525d-9b5f-48ae-86cc-13e72c74d29f)

I am currently developing a straightforward graphical user interface (GUI) project with the following features:

- Enabling users to add and remove robotic arms.
- Automatic scene initialization and maintenance.


Despite the simplicity of the concept, I am encountering several challenges during the development process, particularly concerning cross-platform compatibility and dependency management.
Join Discord for collaborative gui-development: https://discord.gg/5XgsQaBhxX

**Reply by 00make** [reply]
On macOS, I’ve encountered issues with separating rendering and simulation, which requires using run_in_another_thread to ensure smooth operation. This threading approach is necessary to prevent the UI from freezing during simulation, but managing dependencies across different platforms, especially macOS, has proven to be quite challenging.(光滑的) to ensure smooth operation

**Comment by zhouxian** [comment]
Wow I feel so encouraged to see enthusiastic people like you helping us!!! The general design principle would be try to make it lightweight and intuitive, and looking good (if possible :P) let me know if you need any support from us!

**Comment by OIEIEIO** [comment]
Great setup, works like a charm (ubuntu 24 | 2-e52699v3 | 256Gb ram | 4060ti 16gb)
![Screenshot from 2024-12-22 09-53-48](https://github.com/user-attachments/assets/4c5f1118-eb1c-431a-aa4c-dff517ca3051)

**Comment by freezeLUO** [comment]
I support your idea! A well-designed GUI to help people with initial setup and configuration is definitely necessary.

**Comment by 00make** [comment]
Before starting gui development, we need to set up docker or virtual environment to adapt to different devices, which is a necessary step.

**Reply by JocelynVelarde** [reply]
Well then, we should get going with something initial. Perhaps as someone mentioned earlier, we should try to adapt it into a simple Streamlit dashboard

**Reply by W1nson** [reply]
I'm interested in working together on it

**Reply by JocelynVelarde** [reply]
> I'm interested in working together on it

Is Streamlit a good starting point for this?

**Reply by W1nson** [reply]
yes

**Reply by JocelynVelarde** [reply]
let me open a discussion to see if anyone else wants to be on board with this idea

**Reply by Excidos** [reply]
https://github.com/Excidos/Genesis-Ollama.git

I've been working on a really simple Streamlit UI for Gensis with Ollama as the driving creator of the scene. Obviously I'm no programming, Claude helped me but if you want to use any of the functionality feel free to :)

**Reply by JocelynVelarde** [reply]
hey awesome! i mean we could definitely add more pages and sections to the streamlit

**Reply by Excidos** [reply]
Yeah of course, whether you just want to use the code, or if I can add you and other as editors to the repository. I'm a bit too busy in this next month to focus too much on it I'm afraid, whatever works for everyone :)

**Reply by JocelynVelarde** [reply]
Yes ofc go ahead and add me as contributor and I'll start coding perhaps after holidays 😵‍💫

**Reply by W1nson** [reply]
Yeah please add me as well. Thank you I can get started

**Reply by JocelynVelarde** [reply]
can i get some discord usernames so we can agree on things to work on faster?

**Reply by Excidos** [reply]
indrasmirror (Discord username)

**Reply by W1nson** [reply]
w1nny is my dc username

**Reply by JocelynVelarde** [reply]
done!

**Reply by Ginogeo** [reply]
I don't know much about this , but is integrating this with blender an option.
